modules/sort/main/index.coffee

```diff
@@ -1,15 +1,17 @@
 # Intl.Collator isn't supported by safari
 # https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Collator
 # using a collator is supposed to be faster than doing localeCompare
-hasCollator = Intl?.Collator?
+hasCollator = -> Intl?.Collator?
 
-collator = if hasCollator
+collatorFn = -> if hasCollator()
   new Intl.Collator(undefined, {numeric: true}).compare
 else
   (a, b) ->
-    if a < b then -1
-    else if a > b then 1
-    else 0
+    if a is b then 0
+    else if String(a) > String(b) then 1
+    else -1
+    
+collator = if (window.chai?) then (a, b) -> collatorFn()(a, b) else collatorFn()
```

Comment:
We could get around the need to do this by storing collator as `hx._.sort.collator` and deleting it in the tests.
I'm not keen on adding code to hexagon to support tests...

e.g.
`hx._.sort.compare = collatorFn()`

Use `hx._.sort.compare(a, b)` in `compare`

Then in tests you can do `delete hx._.sort.compare` after removing `Intl`

This way it's agnostic of the way the tests are implemented and still retains the 'singleton' behaviour in the browser